### PR TITLE
feat: Add delta constraints to better ensure result data

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## Description

Included in this PR
- Removed responsibility of creating delta table from process step result writer to migration
- Removed all existing migrations as they are no longer needed
- Removed all migration tests
- Replaced migration tests with production code tests

This brings the concept of delta table migrations - and the testing - very close to the principles from the .NET SQL migrations.

## References

* https://app.zenhub.com/workspaces/mandalorian-5fd22a1a59e4740018f5ab5a/issues/gh/energinet-datahub/green-energy-hub/415

### CODE REVIEW

File: source/databricks/calculation-engine/tests/datamigration/migration_scripts/test_202304130900_Add_result_table_constraints.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        ("grid_area", "1234"),
+        ("out_grid_area", "12"),
+        ("out_grid_area", "1234"),
+        ("batch_process_type", "foo"),
+        ("time_series_type", "foo"),
+        ("quantity_quality", "foo"),
+        ("aggregation_level", "foo"),
+    ],
+)
+def test__apply__enforces_data_constraints(
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/tests/datamigration/migration_scripts/test_202304130900_Add_result_table_constraints.py:42]: Is this more a test of process_step_result_writer? If one day we decide to remove the migration script, we still might want to test constraints

Comment (BjarkeMeier) [reply source/databricks/calculation-engine/tests/datamigration/migration_scripts/test_202304130900_Add_result_table_constraints.py:42]: I'll actually move the responsibility for creating the delta table out of the writer and into a migration.

File: source/databricks/calculation-engine/package/file_writers/process_step_result_writer.py [hunk 2]
```diff
@@ -13,15 +13,12 @@
...
 from pyspark.sql.functions import col, lit
-from pyspark.sql import SparkSession
 
 import package.infrastructure as infra
 from package.codelists import MarketRole, TimeSeriesType, AggregationLevel
-from package.constants import Colname, PartitionKeyName
-from package.schemas import results_schema, ResultSchemaField
+from package.constants import Colname, PartitionKeyName, ResultTableColName
 
 DATABASE_NAME = "wholesale_output"
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/package/file_writers/process_step_result_writer.py:23]: Should we define these variables one place only, or is it better to also hard code it in the migration script?

File: source/databricks/calculation-engine/tests/integration/calculator/calculator_job/conftest.py [hunk 3]
```diff
@@ -56,27 +59,21 @@ def executed_calculation_job(
...
-        # Since we are appending the result dataframes we must ensure that the path is removed before executing the tests
-        rmtree(output_path)
-
     metering_points_df = spark.read.csv(
         f"{test_files_folder_path}/MeteringPointsPeriods.csv",
         header=True,
         schema=metering_point_period_schema,
     )
     metering_points_df.write.format("delta").save(
-        f"{data_lake_path}/{worker_id}/calculation-input-v2/metering-point-periods",
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/tests/integration/calculator/calculator_job/conftest.py:79]: Will this not mean that we can't run in parallel?

File: source/databricks/calculation-engine/package/file_writers/process_step_result_writer.py (hunk 2 above)
Comment (BjarkeMeier) [reply source/databricks/calculation-engine/package/file_writers/process_step_result_writer.py:23]: I believe they are maintained by means of migrations. Thus they need to be hardcoded in the migrations. This is in contrast to e.g. the container path, which I actually moved out of the migrations for the same reason.

**Review by JesperGomes — APPROVED** [review]

**Pull request merged**